safedelete/tests/test_soft_delete.py

```diff
@@ -104,3 +104,33 @@ def test_validate_unique(self):
                 name='test'
             ).validate_unique
         )
+
+    def test_check_unique_fields_exists(self):
+        # No unique fields
+        self.assertEqual(SoftDeleteModel.has_unique_fields(), False)
+        self.assertEqual(UniqueSoftDeleteModel.has_unique_fields(), True)
+
+    def test_update_or_create_no_unique_field(self):
+        SoftDeleteModel.objects.update_or_create(id=1)
+        obj, created = SoftDeleteModel.objects.update_or_create(id=1)
+        self.assertEqual(obj.id, 1)
+
+    def test_update_or_create_with_unique_field(self):
+        # Create and soft-delete object
+        obj, created = UniqueSoftDeleteModel.objects.update_or_create(name='unique-test')
+        obj.delete()
+        # Update it and see if it fails
+        obj, created = UniqueSoftDeleteModel.objects.update_or_create(name='unique-test')
+        self.assertEqual(obj.name, 'unique-test')
+        self.assertEqual(created, False)
+
+    def test_update_or_create_flag_with_settings_flag_active(self):
+        self.modify_settings(SAFE_DELETE_INTERPRET_UNDELETED_OBJECTS_AS_CREATED=True)
```

Comment:
I think you need to use this as a context manager (with `with`) or you could affect other tests.

Reply:
@Gagaro Sorry, I cant follow you. What do you want me to do?

Reply:
Use the following instead:

```python
with self.modify_settings(SAFE_DELETE_INTERPRET_UNDELETED_OBJECTS_AS_CREATED=True):
    # Only this code will run with these modified settings.
```

You could also use [`override_settings`](https://docs.djangoproject.com/en/2.0/topics/testing/tools/#django.test.override_settings).

Reply:
Yes one of those.

Reply:
@AndreasBackx @Gagaro Sorry for the delay. It's on the way. I used the decorator.

Reply:
The test is failing. I think you should use `override_settings` because the setting is not a dictionary.

Reply:
Thanks. Pushed again.